cpp/src/arrow/flight/types.cc

```diff
@@ -370,10 +397,141 @@ bool FlightEndpoint::Equals(const FlightEndpoint& other) const {
   return ticket == other.ticket && locations == other.locations;
 }
 
+arrow::Result<std::string> FlightEndpoint::SerializeToString() const {
+  pb::FlightEndpoint pb_flight_endpoint;
+  RETURN_NOT_OK(internal::ToProto(*this, &pb_flight_endpoint));
```

Comment:
This doesn't build either (the function is defined, but isn't declared in the header)

Reply:
add the needed header in serialization_internal.h